### ISSUE TITLE
GUI: Graphical Anomalies on Channel Selection

### ISSUE THREAD

**Issue by davecgh** [companyzero/bisonrelay#144]
With version 0.1.4, I've noticed a couple of graphical anomalies.

1) Whenever you click on the "Chats" section to the left, it improperly clears the contents of the actively selected channel and also scrolls the channel to the top.  When you scroll the channel list back down, you can see the originally selected channel is still selected, but you have to click on the channel to get the contents back anyway
2) Whenever you click on a second channel and click back on the original channel when that original channel is full of content, it does some type of extremely distracting top-to-bottom wipe transition animation

I've attached a screen grab of both points, although it might be a little bit hard to follow due to the obfuscation I added for privacy purposes.  The key points are that I'm starting with the "random" channel selected which required scrolling the channel list down to bring it into view.  Then I click on "Chats" and you can see both the content disappear as well as the channel list automatically scroll back to the top.  Then, I scroll back down to show "random" is still highlighted followed by clicking on the "random" channel again to bring the contents back.  Finally, I click on the "bisonrelay" channel and click back on the "random" channel to show the aforementioned strange transition animation.

![bruig_channel_selection_anomalies](https://user-images.githubusercontent.com/2115102/220266520-68cf17f5-461b-465b-9ff3-26fa3f7d287d.gif)

**Comment by alexlyp** [comment]
I've been able to fix the first issue pretty easily, though the second issue is causing a bit more an issue.  While we can retain scroll position between clicks, I'm struggling to figure out a good way to avoid seeing the scroll to position animation.

**Comment by davecgh** [comment]
I believe what is causing it is that it appears to be reloading the text widget with all of the content and then setting its position to the bottom.

I highly suspect this because I noticed on the huge utf-8 test that I did in the bisonrelay channel means that every time I click on that channel in the GUI, it takes quite a while (as in several seconds) to load everything before it finally jumps to the bottom.

Also, I've noticed that as the amount of data in each channel grows, the effect becomes much more pronounced.

Given that, I expect a good fix (and one I've seen used in other chat-style software) is to only load a fixed amount of history and dynamically load more as the user scrolls up to the top until there is no more to load.
